### PR TITLE
Add metadata to local_channels table

### PR DESCRIPTION
**Pull request by pm47**
(First commit is a refactor/cleanup and can be reviewed independently).

Here is the rationale for implementing channel metadata as additional
columns in the `local_channels` table of the `channels` db, as opposed
to a dedicated `channel_metadata` table of a `audit` db:

1) There is a migration to do (in the `local_channels` table, no less!),
but it's just a table migration, as opposed to a data migration, if we
had to populate a new table in a separate database.
2) We don't need to worry about creating a new metadata line when a new
channel is created (compared to doing add-or-update stuff). It's only
_updating_ optional columns in a best-effort manner.
3) We don't need to worry about inconsistencies between two tables
located in two separated databases (that's a big one).
4) We may want to use the metadata during operations, not just for
audit purposes. For example to close channels that have stayed unused
for a long time.
5) The audit db is an append-only log of events and shouldn't be used
for anything else. There is no `UPDATE` sql statement in
`*AuditDb.scala`. The `channel_metadata` would break that heuristic.

### CODE REVIEW

**Comment by pm47** [comment]
@sstone @t-bast the PR isn't finished but I'd like to have your feedback before I go further. I did start by implementing this in the `audit` db, before changing for this approach.

**Comment by t-bast** [comment]
That sounds reasonable, these are good reasons to go down that route.
As long as we use separate, optional columns and not modify the core `ChannelData` structures let's try and see where it takes us.

File: eclair-core/src/main/scala/fr/acinq/eclair/db/DbEventHandler.scala [hunk 1]
```diff
@@ -65,49 +74,68 @@ class Auditor(nodeParams: NodeParams) extends Actor with ActorLogging {
           PaymentMetrics.PaymentParts.withTag(PaymentTags.Direction, PaymentTags.Directions.Sent).record(outgoing.length)
         case _: ChannelPaymentRelayed =>
       }
-      db.add(e)
+      auditDb.add(e)
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/db/DbEventHandler.scala:77]: Why don't you update the channel meta here as well?

File: eclair-core/src/main/scala/fr/acinq/eclair/db/DbEventHandler.scala [hunk 2]
```diff
@@ -65,49 +74,68 @@ class Auditor(nodeParams: NodeParams) extends Actor with ActorLogging {
...
       e match {
         case ChannelStateChanged(_, channelId, _, remoteNodeId, WAIT_FOR_FUNDING_LOCKED, NORMAL, Some(commitments: Commitments)) =>
           ChannelMetrics.ChannelLifecycleEvents.withTag(ChannelTags.Event, ChannelTags.Events.Created).increment()
-          db.add(ChannelLifecycleEvent(channelId, remoteNodeId, commitments.capacity, commitments.localParams.isFunder, !commitments.announceChannel, "created"))
+          val event = ChannelLifecycleEvent.EventType.Created
+          auditDb.add(ChannelLifecycleEvent(channelId, remoteNodeId, commitments.capacity, commitments.localParams.isFunder, !commitments.announceChannel, event))
+          channelsDb.updateChannelMeta(channelId, event)
         case ChannelStateChanged(_, _, _, _, WAIT_FOR_INIT_INTERNAL, _, _) =>
+        case ChannelStateChanged(_, channelId, _, _, OFFLINE, SYNCING, _) =>
+          channelsDb.updateChannelMeta(channelId, ChannelLifecycleEvent.EventType.Connected)
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/db/DbEventHandler.scala:99]: We may want to have this event in the `auditDb` as well, don't we?

File: eclair-core/src/test/scala/fr/acinq/eclair/db/SqliteChannelsDbSpec.scala [hunk 3]
```diff
@@ -69,7 +73,64 @@ class SqliteChannelsDbSpec extends AnyFunSuite {
...
+      db.updateChannelMeta(channel1.channelId, ChannelLifecycleEvent.EventType.Closed(null))
+      assert(getTimestamp(channel1.channelId, "closed_timestamp").nonEmpty)
+
+      // make sure all metadata are still empty for channel 2
+      assert(getTimestamp(channel2.channelId, "created_timestamp").isEmpty)
+      assert(getTimestamp(channel2.channelId, "last_payment_sent_timestamp").isEmpty)
+      assert(getTimestamp(channel2.channelId, "last_payment_received_timestamp").isEmpty)
+      assert(getTimestamp(channel2.channelId, "last_connected_timestamp").isEmpty)
+      assert(getTimestamp(channel2.channelId, "closed_timestamp").isEmpty)
+
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/db/SqliteChannelsDbSpec.scala:129]: nit: empty line

File: eclair-core/src/main/scala/fr/acinq/eclair/db/DbEventHandler.scala (hunk 2 above)
Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/db/DbEventHandler.scala]: I don't think so, because the risk of spam is too high imo.

**Pull request merged**